vslam/tracker.py

```diff
@@ -193,39 +195,43 @@ def matchImagePair(
             Tuple containing 2D matched keypoints from database and query image respectively
 
         """
-        # 1. Detect keypoints in image 1 (prev) and 2 (curr)
-        prev_image_gray = cv2.cvtColor(prev_image, cv2.COLOR_BGR2GRAY)
-        curr_image_gray = cv2.cvtColor(curr_image, cv2.COLOR_BGR2GRAY)
-
-        prev_corners = cv2.goodFeaturesToTrack(prev_image_gray,
-                                               self.config["num_interest_points"],
-                                               self.config["quality_level"],
-                                               self.config["nms_radius"])
-        #TODO Visualize prev_keypoints for logger in verbose debug mode
-        # prev_image_corners = prev_image.copy()
-        # cv2.waitKey(0)
-        # for corner in prev_corners:
-        #     x, y = corner.ravel()
-        #     cv2.drawMarker(prev_image_corners, (x, y), (0, 255, 0), markerType=cv2.MARKER_CROSS, thickness=1, markerSize=10)
-        # cv2.imshow("Corners debug", prev_image_corners)
-        # cv2.waitKey(0)
-
-        curr_corners = cv2.goodFeaturesToTrack(curr_image_gray,
-                                               self.config["num_interest_points"],
-                                               self.config["quality_level"],
-                                               self.config["nms_radius"])
-
-        prev_keypoints = _corners_to_keypoints(prev_corners)
-        curr_keypoints = _corners_to_keypoints(curr_corners)
-
-        # 2. Compute descriptors for detected features
-        orb = cv2.ORB_create(self.config["num_interest_points"],
-                             scaleFactor=1.2,
-                             nlevels=3)
-        prev_keypoints, prev_desc = orb.compute(prev_image_gray,
-                                                prev_keypoints)
-        curr_keypoints, curr_desc = orb.compute(curr_image_gray,
-                                                curr_keypoints)
+        # If feature provided
+        if self.feature is not None:
+            prev_keypoints, prev_desc = self.feature.detectAndCompute(prev_image)
+            curr_keypoints, curr_desc = self.feature.detectAndCompute(curr_image)
+        else:
+            # 1. Detect keypoints in image 1 (prev) and 2 (curr)
```

Comment:
else use a default feature extractor say KLTFeatureExtractor? and move this code there?